python/trezorlib/binance.py

```diff
@@ -0,0 +1,114 @@
+from . import messages
+from .tools import expect, session
+
+
+@expect(messages.BinanceAddress, field="address")
+def get_address(client, address_n, show_display=False):
+    return client.call(
+        messages.BinanceGetAddress(address_n=address_n, show_display=show_display)
+    )
+
+
+@expect(messages.BinancePublicKey, field="public_key")
+def get_public_key(client, address_n, show_display=False):
+    return client.call(
+        messages.BinanceGetPublicKey(address_n=address_n, show_display=show_display)
+    )
+
+
+@session
+def sign_tx(client, address_n, tx_json):
+    msg = tx_json["msgs"][0]
+    envelope = messages.BinanceSignTx(
+        msg_count=1,
+        account_number=int(tx_json["account_number"]),
+        chain_id=tx_json["chain_id"],
+        memo=tx_json["memo"],
+        sequence=int(tx_json["sequence"]),
+        source=int(tx_json["source"]),
+        address_n=address_n,
+    )
+
+    response = client.call(envelope)
+
+    if not isinstance(response, messages.BinanceTxRequest):
+        raise RuntimeError(
+            "Invalid response, expected BinanceTxRequest, received "
+            + type(response).__name__
+        )
+
+    msg_type = determine_message_type(msg)
```

Comment:
please either make the `determine_message_type` return the `Binance*Msg` directly, or inline the ifs from `determine_message_type` here